### PR TITLE
Optimising the support tests

### PR DESCRIPTION
**Pull request by phyzical**
(ignore the failing tests for now)
This was from a run reducing just the ETL tests, and there is still improvements that could be done by chaining .and most are just merged it blocks where context is no different between tests.

The one downside to this change is the context/it blocks that are removed that may provide more information to a dev.

While this time taken might not feel like much, it would proably get better by performing such adjustments across the whole test suite, also locally for some reason the entire suite can take ~25 minutes on my machine so hoping these changes improve this also

Before:
Finished in 4 minutes 11.3 seconds (files took 4.85 seconds to load)
2098 examples, 0 failures
After:
Finished in 3 minutes 46.7 seconds (files took 6.48 seconds to load)
1654 examples, 0 failures

### CODE REVIEW

File: spec/support/jobs/concerns/etl/common.rb [hunk 1]
```diff
@@ -118,95 +118,59 @@ def description
...
+      ).and(
+        change { subject.reload.ui_info.descriptions.first.value }.to(description.squish)
+      ))
     end
   end
 
   shared_examples 'updating an SSODescriptor' do
     include_examples 'updating a RoleDescriptor'
 
     it 'updates name id formats' do
```
Comment (matthew-puku) [inline spec/support/jobs/concerns/etl/common.rb:166]: I think it'd be worthwhile to update these descriptions to reflect the broader scope of the assertions. I see you have done that in some places already so maybe you were intending to come back to these anyway.

Comment (phyzical) [reply spec/support/jobs/concerns/etl/common.rb:166]: yeah  i agree, and your right i just left it alone to get some benchmark timings before putting the effort

File: spec/support/jobs/concerns/etl/common.rb [hunk 2]
```diff
@@ -118,95 +118,59 @@ def description
...
-    end
-
-    it 'updates artifact resolution services' do
-      expect { run }.to(change { subject.reload.artifact_resolution_services })
-    end
-
-    it 'updates single_logout_services' do
-      expect { run }.to(change { subject.reload.single_logout_services })
-    end
-
```
[the remark this answers is not in the document]
Comment (ArthurZheng) [reply spec/support/jobs/concerns/etl/common.rb:207]: 👍 ++

File: spec/support/jobs/concerns/etl/attribute_authorities.rb [hunk 3]
```diff
@@ -312,18 +305,10 @@ def run
       end
 
       context 'scopes' do
-        it 'sets a scope' do
+        it 'sets scopes and regex to false' do
```
Comment (matthew-puku) [inline spec/support/jobs/concerns/etl/attribute_authorities.rb:308]:
```suggestion
        it 'sets scope and sets regex to false' do
```

**Review by matthew-puku — CHANGES_REQUESTED** [review]
will work thru now and nit the it blocks. Shouldn't be long

File: spec/support/jobs/concerns/etl/endpoints.rb [hunk 4]
```diff
@@ -1,40 +1,30 @@
 # frozen_string_literal: true
 
 RSpec.shared_examples 'endpoint' do
-  it 'has source data' do
+  it 'expected data' do
```
Comment (matthew-puku) [inline spec/support/jobs/concerns/etl/endpoints.rb:4]:
```suggestion
  it 'sets expected data' do
```

File: spec/support/jobs/concerns/etl/endpoints.rb [hunk 5]
```diff
@@ -1,40 +1,30 @@
...
   include_examples 'endpoint'
 
-  it 'sets is_default' do
-    source.each_with_index do |s, i|
-      expect(target[i].is_default).to eq(s.is_default)
-    end
-  end
-
-  it 'sets index' do
+  it 'expected data' do
```
Comment (matthew-puku) [inline spec/support/jobs/concerns/etl/endpoints.rb:21]:
```suggestion
  it 'sets expected data' do
```

File: spec/support/jobs/concerns/etl/identity_providers.rb [hunk 6]
```diff
@@ -51,7 +51,7 @@ def create_json(idp)
       }
     }
   end
-  # rubocop:enable Metrics/MethodLength,Metrics/AbcSize
+  # rubocop:enable  Metrics/MethodLength,Metrics/AbcSize
```
Comment (matthew-puku) [inline spec/support/jobs/concerns/etl/identity_providers.rb:54]:
```suggestion
  # rubocop:enable Metrics/MethodLength,Metrics/AbcSize
```

File: spec/support/jobs/concerns/etl/identity_providers.rb [hunk 7]
```diff
@@ -205,18 +205,10 @@ def run
       end
 
       context 'scopes' do
-        it 'sets a scope' do
+        it 'sets a scope and refex to false' do
```
Comment (matthew-puku) [inline spec/support/jobs/concerns/etl/identity_providers.rb:208]:
```suggestion
        it 'sets a scope with regexp: false' do
```

File: spec/support/jobs/concerns/etl/identity_providers.rb [hunk 8]
```diff
@@ -331,43 +323,24 @@ def run
     include_examples 'updating an SSODescriptor'
     include_examples 'updating MDUI content'
 
-    it 'uses the existing instance' do
-      expect { run }.not_to(change { IDPSSODescriptor.count })
-    end
-
-    it 'sets a scope' do
+    it 'Works as expected, sets scope, doesnt make tags, updates sso, assertion id, attribute profile and attributes' do
```
Comment (matthew-puku) [inline spec/support/jobs/concerns/etl/identity_providers.rb:326]:
```suggestion
    it 'works as expected, sets scope, doesnt make tags, updates sso, assertion id, attribute profile and attributes' do
```

File: spec/support/jobs/concerns/etl/service_providers.rb [hunk 9]
```diff
@@ -391,9 +383,5 @@ def run
         expect { run }.not_to(change { subject.reload.discovery_response_services })
       end
     end
-
-    it 'updates attribute_consuming_services' do
```
Comment (matthew-puku) [inline spec/support/jobs/concerns/etl/service_providers.rb:395]: lol why was this down here

File: spec/support/metadata/contact_person.rb [hunk 10]
```diff
@@ -32,10 +27,8 @@
       end
     end
     context 'when known' do
-      it 'is created' do
+      it 'is created has correct value' do
```
Comment (matthew-puku) [inline spec/support/metadata/contact_person.rb:30]:
```suggestion
      it 'is created with correct value' do
```

File: spec/support/metadata/contact_person.rb [hunk 11]
```diff
@@ -50,10 +43,8 @@
...
```
Comment (matthew-puku) [inline spec/support/metadata/contact_person.rb:46]:
```suggestion
      it 'is created with correct value' do
```

File: spec/support/metadata/contact_person.rb [hunk 12]
```diff
@@ -68,10 +59,8 @@
...
```
Comment (matthew-puku) [inline spec/support/metadata/contact_person.rb:62]:
```suggestion
      it 'is created with correct value' do
```

File: spec/support/metadata/contact_person.rb [hunk 13]
```diff
@@ -86,10 +75,8 @@
...
```
Comment (matthew-puku) [inline spec/support/metadata/contact_person.rb:78]:
```suggestion
      it 'is created with correct value' do
```

File: spec/support/metadata/contact_person.rb [hunk 14]
```diff
@@ -105,11 +92,9 @@
...
```
Comment (matthew-puku) [inline spec/support/metadata/contact_person.rb:95]:
```suggestion
      it 'is created with correct value' do
```

File: spec/support/metadata/endpoint.rb [hunk 15]
```diff
@@ -7,30 +7,16 @@
...
-      end
-      context 'when populated' do
-        let(:endpoint) { create parent_node, :response_location }
-        it 'is rendered' do
-          expect(xml).to have_xpath("#{endpoint_path}[@ResponseLocation]")
-        end
-        it 'has expected value' do
-          expect(node['ResponseLocation']).to eq(endpoint.response_location)
-        end
+    context 'ResponseLocation when populated' do
```
Comment (matthew-puku) [inline spec/support/metadata/endpoint.rb:15]:
```suggestion
    context 'when ResponseLocation is populated' do
```

File: spec/support/metadata/endpoint.rb [hunk 16]
```diff
@@ -7,30 +7,16 @@
...
-    context 'Binding' do
-      it 'has expected value' do
-        expect(node['Binding']).to eq(endpoint.binding)
-      end
-    end
-    context 'Location' do
-      it 'has expected value' do
-        expect(node['Location']).to eq(endpoint.location)
-      end
+    it 'has binding, location' do
```
Comment (matthew-puku) [inline spec/support/metadata/endpoint.rb:10]:
```suggestion
    it 'has binding and location' do
```

File: spec/support/metadata/endpoint.rb [hunk 17]
```diff
@@ -7,30 +7,16 @@
...
-        let(:endpoint) { create parent_node, :response_location }
-        it 'is rendered' do
-          expect(xml).to have_xpath("#{endpoint_path}[@ResponseLocation]")
-        end
-        it 'has expected value' do
-          expect(node['ResponseLocation']).to eq(endpoint.response_location)
-        end
+    context 'ResponseLocation when populated' do
+      let(:endpoint) { create parent_node, :response_location }
+      it 'is rendered, location' do
```
Comment (matthew-puku) [inline spec/support/metadata/endpoint.rb:17]:
```suggestion
      it 'is rendered and has location' do
```

File: spec/support/metadata/entities_descriptor.rb [hunk 18]
```diff
@@ -29,32 +29,23 @@
...
+      expect(xml).to have_xpath(all_publication_infos, count: 1)
     end
 
     context 'attributes' do
       let(:node) { xml.find(:xpath, entities_descriptor_path) }
 
       around { |example| Timecop.freeze { example.run } }
 
-      it 'sets ID' do
+      it 'sets ID and name and validuntil' do
```
Comment (matthew-puku) [inline spec/support/metadata/entities_descriptor.rb:42]:
```suggestion
      it 'sets ID, name and validuntil' do
```

File: spec/support/metadata/mdui_ui_info.rb [hunk 19]
```diff
@@ -11,21 +11,17 @@
...
 
   context 'DisplayName' do
-    it 'is created' do
-      expect(xml).to have_xpath(mdui_display_name_path, count: 1)
-    end
-
     context 'rendered node' do
       let(:node) { xml.first(:xpath, mdui_display_name_path) }
-      it 'sets language' do
+      it 'sets languageand text' do
```
Comment (matthew-puku) [inline spec/support/metadata/mdui_ui_info.rb:23]:
```suggestion
      it 'sets language and text' do
```

File: spec/support/metadata/mdui_ui_info.rb [hunk 20]
```diff
@@ -130,11 +110,9 @@
 
     context 'rendered node' do
       let(:node) { xml.first(:xpath, mdui_privacystatementurl_path) }
-      it 'sets language' do
+      it 'sets language, text' do
```
Comment (matthew-puku) [inline spec/support/metadata/mdui_ui_info.rb:113]:
```suggestion
      it 'sets language and text' do
```

File: spec/support/metadata/signature.rb [hunk 21]
```diff
@@ -14,50 +14,30 @@
...
-    expect(signed_info).to have_xpath('ds:SignatureMethod', count: 1)
-  end
-
-  it 'includes the reference element' do
-    expect(signed_info).to have_xpath('ds:Reference', count: 1)
-  end
-
-  it 'designates the root element to be signed' do
-    expect(reference['URI']).to eq("##{subject.instance_id}")
-  end
```
Comment (matthew-puku) [inline spec/support/metadata/signature.rb:40]: this one is pretty cryptic. Without the it description I have no idea what this assertion is for. Maybe it should go back. Or you could put a comment within the block

File: spec/support/metadata/sp_sso_descriptor.rb [hunk 22]
```diff
@@ -15,68 +15,44 @@
...
-      it 'is not rendered' do
-        expect(node['AuthnRequestsSigned']).to be_falsey
-      end
-    end
-
-    context 'WantAssertionsSigned' do
-      it 'is not rendered' do
-        expect(node['WantAssertionsSigned']).to be_falsey
-      end
+    it 'is not rendered' do
```
Comment (matthew-puku) [inline spec/support/metadata/sp_sso_descriptor.rb:26]:
```suggestion
    it 'are not rendered' do
```

File: spec/support/metadata/sso_role_descriptor.rb [hunk 23]
```diff
@@ -14,75 +14,49 @@
     "#{sso_descriptor_path}/NameIDFormat"
   end
 
-  it 'is created' do
+  it 'is created without sso, name_id and artificat resolution' do
```
Comment (matthew-puku) [inline spec/support/metadata/sso_role_descriptor.rb:17]:
```suggestion
  it 'is created without sso, name_id and artifact resolution' do
```

**Review by matthew-puku — APPROVED** [review]
Much more readable overall already tbh but may as well fix the typos

File: spec/support/metadata/signature.rb (hunk 21 above)
Comment (phyzical) [reply spec/support/metadata/signature.rb:40]: added some inline comments to try to preserve some of the it blocks futher

**Pull request merged**